### PR TITLE
Implements Exit Numbers + Names

### PR DESCRIPTION
**Pull request by daniel-j-h**
For https://github.com/Project-OSRM/osrm-backend/issues/3400: implements exit numbers and exit names using the osm `junction:ref` way tag for now. Exposes a `exits` property in the response's `RouteStep`. 

- [x] Measure increase in memory consumption
- [x] Review
- [x] Adjust for comments

cc @miccolis @willwhite @danpat

### CODE REVIEW

File: include/util/guidance/name_announcements.hpp [hunk 1]
```diff
@@ -129,16 +131,34 @@ inline bool requiresNameAnnounced(const std::string &from_name,
...
+    //         ` e . . f
+    //
+    // could assign the exit number to `be` when exiting `abcd` instead of the full ramp.
+    //
+    // Issuing a new-name instruction here would result in the turn assuming the short segment to be
+    // irrelevant and remove the exit number in a collapse scenario. We don't want to issue any
+    // instruction from be-ef, since we only loose the exit number. So we want to make sure that we
+    // don't just loose an exit number, when exits change
+    const auto exits_change = from_exits != to_exits;
+    const auto looses_exit = (names_are_equal && !from_exits.empty() && to_exits.empty());
```
Comment (TheMarex) [inline include/util/guidance/name_announcements.hpp]: I'm unclear on what the overall effect here is supposed to be, please clarify on which these are true:

1/ We emit a new-name instruction at `be->ef`
2/ We don't emit a new-name instruciton at `be->ef`
3/ We just ensure these steps won't be collapsed
4/ We just ensure the exit number will be copied to the new step

Comment (MoKob) [inline include/util/guidance/name_announcements.hpp]: The overall effect is 2), not announcing a new name. If we would announce a new name, the engine would see a short segment after which a new name is given and use the second name for the ramp instruction.
By not announcing a new name, we ensure that the name with the exit number is used for the ramp.

**Comment by MoKob** [comment]
Some insight into index change: on Germany, my index increases from 9 MB to 9.2 MB.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,9 @@
       - new parameter `approaches` for `route`, `table`, `trip` and `nearest` requests.  This parameter keep waypoints on the curb side.
         'approaches' accepts both 'curb' and 'unrestricted' values.
         Note : the curb side depend on the `ProfileProperties::left_hand_driving`, it's a global property set once by the profile. If you are working with a planet dataset, the api will be wrong in some countries, and right in others.
+      - Support for exits numbers and names. New member `exits` in `RouteStep`, based on `junction:ref` on ways
```
Comment (TheMarex) [inline CHANGELOG.md]: Just noticed that this was added on the `5.8` entry, not a `5.9` one.

**Comment by TheMarex** [comment]
Urg I think travis updated eslint on the OSX image, I think the failure is unrelated. Needs a fix on `master`.

**Pull request merged**